python/dgl/data/csv_dataset.py

```diff
@@ -0,0 +1,420 @@
+        csv_path = file_name
+        if base_dir is not None:
+            csv_path = os.path.join(base_dir, csv_path)
+        return pd.read_csv(csv_path, sep=separator)
+
+    @staticmethod
+    def load_node_data_from_csv(meta_node, data_parser, base_dir=None, separator=','):
+        df = CSVDataLoader.read_csv(meta_node.file_name, base_dir, separator)
+        ntype = meta_node.ntype
+        ndata = data_parser(df)
+        if meta_node.node_id_field not in ndata:
+            raise DGLError(f"node_id_field[{meta_node.node_id_field}] does not exist in parsed node data dict.")
+        node_ids = ndata[meta_node.node_id_field]
+        ndata.pop(meta_node.node_id_field)
+        if meta_node.graph_id_field in ndata:
+            graph_ids = ndata[meta_node.graph_id_field]
+            ndata.pop(meta_node.graph_id_field)
+        else:
+            graph_ids = None
+        return NodeData(node_ids, ndata, type=ntype, graph_id=graph_ids)
+
+    @staticmethod
+    def load_edge_data_from_csv(meta_edge, data_parser, base_dir=None, separator=','):
+        df = CSVDataLoader.read_csv(meta_edge.file_name, base_dir, separator)
+        etype = tuple(meta_edge.etype)
+        edata = data_parser(df)
+        if meta_edge.src_id_field not in edata:
+            raise DGLError(f"src_id_field[{meta_edge.src_id_field}] does not exist in parsed edge data dict.")
+        if meta_edge.dst_id_field not in edata:
+            raise DGLError(f"dst_id_field[{meta_edge.dst_id_field}] does not exist in parsed edge data dict.")
+        src_ids = edata[meta_edge.src_id_field]
+        dst_ids = edata[meta_edge.dst_id_field]
+        graph_ids = edata[meta_edge.graph_id_field] if meta_edge.graph_id_field in edata else None
+        for key in [meta_edge.src_id_field, meta_edge.dst_id_field, meta_edge.graph_id_field]:
+            if key in edata:
+                edata.pop(key)
+        return EdgeData(src_ids, dst_ids, edata, type=etype, graph_id=graph_ids)
+
+    @staticmethod
+    def load_graph_data_from_csv(meta_graph, data_parser, base_dir=None, separator=','):
+        df = CSVDataLoader.read_csv(meta_graph.file_name, base_dir, separator)
+        gdata = data_parser(df)
+        if meta_graph.graph_id_field not in gdata:
+            raise DGLError(f"graph_id_field[{meta_graph.graph_id_field}] does not exist in parsed graph data dict.")
+        graph_ids = gdata[meta_graph.graph_id_field]
+        gdata.pop(meta_graph.graph_id_field)
+        return GraphData(graph_ids, gdata)
+
+
+class DGLGraphConstructor:
+    """ Class for constructing DGLGraph from Node/Edge/Graph data. Internal use only. """
+    @staticmethod
+    def construct_graphs(node_data, edge_data, graph_data=None):
+        if not isinstance(node_data, list):
+            node_data = [node_data]
+        if not isinstance(edge_data, list):
+            edge_data = [edge_data]
+        node_dict = DGLGraphConstructor._parse_node_data(node_data)
```

Comment:
I will also suggest moving this method to `NodeData` as a member such as `NodeData.to_data_dict()`.

Reply:
accepted.